### PR TITLE
Add ParticleMeshCounter Warp utility for particle-in-mesh counting

### PR DESCRIPTION
**Pull request by maxkra15**
# Description

This PR adds `isaaclab.utils.warp.ParticleMeshCounter`, a fast, solver-agnostic Warp utility for counting
how many particles fall inside one or more closed (watertight) region meshes.

**What's added**

- **`ParticleMeshCounter`** — counts particles inside region meshes using Warp's BVH-accelerated
  winding-number point query (`wp.mesh_query_point_sign_winding_number`). It supports multiple,
  independently posed region meshes per environment (region geometry is fixed in its local frame and the
  BVH is built once and reused; only per-environment transforms change between calls), sanitizes
  non-finite particle positions, and returns per-region counts and/or the per-particle containment mask.
- **`make_box_region_mesh` / `make_frustum_region_mesh`** — helpers that build watertight, outward-oriented
  region meshes (axis-aligned boxes and capped circular frusta / "cup cavity" regions).

It lives next to the existing Warp helpers (`raycast_mesh`, `convert_to_warp_mesh`) in
`isaaclab.utils.warp`, as a single cohesive `particle_mesh.py` module (mirroring the package's existing
feature modules such as `proxy_array.py`).

**Motivation / context**

This enables training-time / privileged measurements such as "how many media particles are inside the
scoop bowl / source container / target container" without relying on hand-tuned analytic regions. It is
pure Warp + Torch with no physics-backend dependency, so it works for any particle source (MPM, PhysX
particles, point clouds).

**Dependencies**

None beyond what Isaac Lab already ships (Warp + Torch).

Fixes # (N/A — no tracking issue)

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshots

N/A — non-visual utility (validated by unit tests against analytic ground truth).

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/utils/warp/particle_mesh.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+from collections.abc import Sequence
+
+import numpy as np
+import torch
+import warp as wp
+
+# Disable Warp's init banner and initialize the module so the kernel below is registered at import
+# time (matches the pattern used in :mod:`isaaclab.utils.warp.ops`).
+wp.config.quiet = True
+wp.init()
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/utils/warp/particle_mesh.py]: the init here might not be necessary

File: source/isaaclab/isaaclab/utils/warp/particle_mesh.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        region_meshes: Sequence[wp.Mesh | tuple[np.ndarray, np.ndarray]],
+        num_envs: int,
+        device: str,
+        *,
+        max_query_dist: float = 1.0e6,
+    ) -> None:
+        if len(region_meshes) == 0:
+            raise ValueError("`region_meshes` must contain at least one region mesh.")
+        self._device = str(device)
+        self._wp_device = wp.device_from_torch(torch.device(self._device))
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/utils/warp/particle_mesh.py]: some of the easily derivable field like `_wp_device`, I usually don't add fields whose semantic is not tightly related to class and is also easily derivable, if user needs it, let user derive themselves

File: source/isaaclab/isaaclab/utils/warp/particle_mesh.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        self._num_envs = int(num_envs)
+        self._max_query_dist = float(max_query_dist)
+        self._meshes: list[wp.Mesh] = [self._as_winding_mesh(mesh) for mesh in region_meshes]
+        self._num_regions = len(self._meshes)
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/utils/warp/particle_mesh.py]: similarity for num_regions

File: source/isaaclab/isaaclab/utils/warp/particle_mesh.py [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+            if region_quat.dim() == 2 and tuple(region_quat.shape) == (self._num_regions, 4):
+                region_quat = region_quat.unsqueeze(1).expand(self._num_regions, self._num_envs, 4)
+            if tuple(region_quat.shape) != (self._num_regions, self._num_envs, 4):
+                raise ValueError(
+                    f"`region_orientations` must broadcast to (num_regions={self._num_regions},"
+                    f" num_envs={self._num_envs}, 4), got {tuple(region_orientations.shape)}."
+                )
+        return region_pos.contiguous(), region_quat.contiguous()
+
+    def _ensure_buffers(self, num_particles: int) -> None:
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/utils/warp/particle_mesh.py]: I see there is only one use-case for _ensure_buffers, usually this type of naming tends to smell because it conveys the intent that `I don't know if buffer is there or not, just ensure its exist` If the behavior is deterministic, change to a more deterministic naming like `initalize_buffers` if the behavior is unknown lets understand the lifecycle better

File: source/isaaclab/isaaclab/utils/warp/particle_mesh.py [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+        self._device = str(device)
+        self._wp_device = wp.device_from_torch(torch.device(self._device))
+        self._num_envs = int(num_envs)
+        self._max_query_dist = float(max_query_dist)
+        self._meshes: list[wp.Mesh] = [self._as_winding_mesh(mesh) for mesh in region_meshes]
+        self._num_regions = len(self._meshes)
+        self._mesh_ids = wp.array([mesh.id for mesh in self._meshes], dtype=wp.uint64, device=self._wp_device)
+        # buffers sized on first use (depend on the per-call particle count)
+        self._num_particles = 0
+        self._inside_torch: torch.Tensor | None = None
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/utils/warp/particle_mesh.py]: inside_torch and inside_warp is a bit weird, are you looking for something like ProxyArray?

**Review by ooctipus — APPROVED** [review]
maybe we can take out of torch from this file all together.
